src/Utility/TextSplitter.cpp

```diff
@@ -0,0 +1,33 @@
+#include "TextSplitter.h"
+#include <iostream>
+int TextSplitter::numOfWindows(long long int textLength, int patternLength) {
+    return textLength - patternLength + 1;
+}
+
+std::pair<int, int> TextSplitter::lenOfChain(int windows) {
+    int rangeWidth = windows / numOfThreads;
+    int rangesWithBonus = windows % numOfThreads;
+    return std::make_pair(rangeWidth, rangesWithBonus - 1);
+}
+
+std::vector<std::tuple<size_t, size_t>> TextSplitter::splitText(long long int textLength, int patternLength) {
+    std::vector<std::tuple<size_t, size_t>> ranges;
+    int windows = numOfWindows(textLength, patternLength);
+    std::pair<int, int> chains = lenOfChain(windows);
+    int currenPosition = 0;
+    for(int i = 0; i < numOfThreads; i++) {
+        if(windows > i) {
+            if (chains.second >= 0) {
+                ranges.push_back(std::make_tuple(currenPosition, currenPosition + chains.first));
```

Comment:
Change `push_back` to `emplace_back` here and below.

Reply:
Right, `emplace_back` seems to be better pick in this place, thanks for advice :smile: